backtrace-library/src/main/java/backtraceio/library/breadcrumbs/BacktraceBreadcrumbs.java

```diff
@@ -0,0 +1,246 @@
+    }
+
+    private void registerAutomaticBreadcrumbReceivers(Set<BacktraceBreadcrumbType> breadcrumbTypesToEnable) {
+        if (isBreadcrumbsEnabled()) {
+            unregisterAutomaticBreadcrumbReceivers();
+        }
+
+        backtraceBroadcastReceiver = new BacktraceBroadcastReceiver(this);
+        context.registerReceiver(backtraceBroadcastReceiver,
+                backtraceBroadcastReceiver.getIntentFilter(breadcrumbTypesToEnable));
+
+        if (breadcrumbTypesToEnable.contains(BacktraceBreadcrumbType.SYSTEM)) {
+            backtraceComponentListener = new BacktraceComponentListener(this);
+            context.registerComponentCallbacks(backtraceComponentListener);
+
+            backtraceActivityLifecycleListener = new BacktraceActivityLifecycleListener(this);
+            if (context instanceof Application) {
+                ((Application) context).registerActivityLifecycleCallbacks(backtraceActivityLifecycleListener);
+            }
+        }
+
+        this.enabledBreadcrumbTypes = breadcrumbTypesToEnable;
+    }
+
+    public void setEnabledBreadcrumbTypes(Set<BacktraceBreadcrumbType> breadcrumbTypesToEnable) {
+        if (breadcrumbTypesToEnable == null)
+            return;
+
+        registerAutomaticBreadcrumbReceivers(breadcrumbTypesToEnable);
+    }
+
+    public void enableBreadcrumbs(Set<BacktraceBreadcrumbType> enabledBreadcrumbTypes)
+    {
+        setEnabledBreadcrumbTypes(enabledBreadcrumbTypes);
+
+        // We should log all breadcrumb configuration changes in the breadcrumbs
+        addConfigurationBreadcrumb();
+    }
+
+    private void unregisterAutomaticBreadcrumbReceivers() {
+        this.context.unregisterReceiver(backtraceBroadcastReceiver);
+        backtraceBroadcastReceiver = null;
+
+        this.context.unregisterComponentCallbacks(backtraceComponentListener);
+        backtraceComponentListener = null;
+
+        if (context instanceof Application) {
+            ((Application) context).unregisterActivityLifecycleCallbacks(backtraceActivityLifecycleListener);
+        }
+    }
+
+    public void disableBreadcrumbs() {
+        unregisterAutomaticBreadcrumbReceivers();
+
+        enabledBreadcrumbTypes = null;
+
+        // We should log all breadcrumb configuration changes in the breadcrumbs
+        addConfigurationBreadcrumb();
+    }
+
+    public boolean clearBreadcrumbs() {
+        boolean success = backtraceBreadcrumbsLogger.clear();
+        // Make sure the configuration is always known
+        addConfigurationBreadcrumb();
+        return success;
+    }
+
+    public boolean isBreadcrumbsEnabled()
+    {
+        return enabledBreadcrumbTypes != null && !enabledBreadcrumbTypes.isEmpty();
+    }
+
+    public static String getBreadcrumbLogDirectory(@NonNull Context context) {
+        return context.getFilesDir().getAbsolutePath() + "/" + BacktraceBreadcrumbs.breadcrumbLogDirectory;
+    }
+
+    public static String getBreadcrumbLogFileName() {
```

Comment:
why do we want to have this method static? Right now if we want to use this method, we know breadcrumbs instance already exist.

Reply:
Is this resolved?

Reply:
Resolved, this method is no longer static in latest code